pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -268,9 +268,9 @@ func (e *etcd) Deploy(ctx context.Context) error {
 						PodSelector: &metav1.LabelSelector{
 							// TODO: Replace below map with a function call to the to-be-introduced kubeapiserver package.
 							MatchLabels: map[string]string{
-								v1beta1constants.DeprecatedGardenRole: v1beta1constants.GardenRoleControlPlane,
-								v1beta1constants.LabelApp:             v1beta1constants.LabelKubernetes,
-								v1beta1constants.LabelRole:            v1beta1constants.LabelAPIServer,
+								v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
```

Comment:
Does this work in one go? I think the network policies are applied earlier, or?

Reply:
The controlplane Pods (except etcds) should have both of the garden role labels since gardener@v1.16 - ref https://github.com/gardener/gardener/pull/3384. So this PR is adapting the NetworkPolicies to use the new garden role label - it is kind of relying that the Shoot was at least once reconciled with gardener >= 1.16.